R/html_document_base.R

```diff
@@ -65,6 +66,23 @@ html_document_base <- function(theme = NULL,
     if (is_bs_theme(theme)) bslib::bs_global_set(old_theme)
   }
 
+  # additional sass/css
+  for (f in css) {
+    if (is_bs_theme(theme)) {
+      theme <- bslib::bs_add_rules(theme, sass::sass_file(f))
+      next
+    }
+    is_sass <- grepl("\\.s[ac]ss$", f)
+    if (is_sass) {
+      f <- sass::sass(
+        sass::sass_file(f),
+        output = sub("\\.s[ac]ss$", ".css", f),
```

Comment:
Just wondering: should the built CSS be consider as a temp file or like here as a file to keep locally once the document is rendered ?

I am fine with this - just wondering if we should think of making it a temp file.

Reply:
Yea, I can't remember if I had a good justification for not using a temp file the 1st time -- I'll go ahead and make that change too.

Reply:
Ok in fact you may not want to do that (or not this way) because it will create a file in temp directory on the environment. This will lead (for a `self_contained: false` document) to creating this in HTML 
```html
<link rel="stylesheet" href="C:\Users\chris\AppData\Local\Temp\RtmpKqPzuP/rmarkdownd9b35e5ea2507976/rmarkdown.min.css" type="text/css">
```

which I believe is not what we want - or at least it is not working. 

I believe this is because you created the temp file in the R temp directory. When using RStudio, the rendering will happen in a new process and R temp folder will be removed at the end of the rendering. 
That is one reason why we don't use `tempdir()` so much in rmarkdown (like for plot file for figures - they are in an intermediary folder). If a dependency file needs to be created for HTML to use, it should be in the `lib_dir` folder (like html dependencies ?) or another intermediates directory. 

Html dependencies have different treatment depending on self contained or not.

So We need to think of that if we want

* the user to provide a `.scss` file
* a `.min.css` is created for Pandoc to use
* no trace of this file is in the working directory of the user. 

If we are ok for the `.css` to be right next to the `.scss`, then we could revert back to your first change. I am not sure leaving a file of a run on user directory is what we want though. Hence my first comment.

Reply:
Ahhh, I guess I was assuming `--css` always inlined the contents because that's all `css` did before I made these changes (the `css` parameter has never worked properly with `self_contained: false`)? I guess I can fix that as well if you'd like

Reply:
> the css parameter has never worked properly with self_contained: false

usually you have your CSS file locally in a relative path to the Rmd file, and so it will stay relative to the html file by default as the output dir is the same directory. Here it will be created in a temp dir which will always be absolute path. 

However you're right, `self_contained: false` + `render(..., output_dir = "out/")` will lead to a css file not copied to the output dir. 

There is a mechanism in `html_document_base` to find absolute references of resource and replace them with relative ones though, but yes it is not working properly currently in the case described the line above. 🤔

Reply:
@cderv I believe 65b0c9d generally addresses the issue?

Reply:
Yes I saw that. I was taken by another issue this morning. I'll have a look now.

Reply:
I just noticed these changes introduce real unit test failures -- would you prefer to go back to the old behavior (i.e., just write the css file to disk? Maybe it'd be possible/safe to remove the css file when `self_contained = TRUE`?)

Reply:
I tried to handle this in ad3f64f